handler.go

```diff
@@ -52,6 +58,22 @@ func (h *handler) Handle(reqURL []string, r *http.Request) *response.Response {
 	}
 }
 
+func (h *handler) verify(ctx []reflect.Value, r *http.Request) error {
+	for _, c := range ctx {
+		guard, isGuard := c.Interface().(Guard)
+		if isGuard {
+			action, err := guard.VerifyRequest(r)
```

Comment:
I think we shouldn't allow the user to get `http.Request`, the user should only get current context value
```
func (u *User) ValidateRequest() action, err {
  if strings.startWith(u.name, "orange") {
    return rocket.Failure, err
  }
  return rocket.Success, nil
}
```

Reply:
Oh, it sounds great WWW, but I think this would need #129 to make it could read `Header` and `Cookie` data.

Reply:
I believe when #129 done, this function can automatically use Header and Cookie.
So you want to work this PR after #129 finish?

Reply:
Yes, I would implement #129 first, would you like to take #122?

Reply:
ok